ganga/GangaCore/Core/GangaRepository/JStreamer.py

```diff
@@ -0,0 +1,196 @@
+"""
+WIP: Replacing the XML files with json equivalent
+
+- JSON Job loader
++ JSON Job dumper
+"""
+
+
+import os
+import json
+
+from GangaCore.Core.exceptions import GangaException
+from GangaCore.Utility.logging import getLogger
+from GangaCore.GPIDev.Base.Proxy import addProxy, stripProxy, isType, getName
+
+from GangaCore.GPIDev.Lib.GangaList.GangaList import GangaList, makeGangaListByRef
+
+# config_scope is namespace used for evaluating simple objects (e.g. File, datetime, SharedDir)
+from GangaCore.Utility.Config import config_scope
+
+from GangaCore.Utility.Plugin import PluginManagerError, allPlugins
+
+from GangaCore.GPIDev.Base.Objects import GangaObject, ObjectMetaclass
+from GangaCore.GPIDev.Schema import Schema, Version
+from GangaCore.GPIDev.Lib.GangaList.GangaList import makeGangaList
+
+# from GangaCore.Core.GangaRepository import SchemaVersionError
+
+import copy
+
+logger = getLogger()
+
+# TODO: Add error handling and logging for the recursion operations 
+class JsonDumper:
+    """Will dump the Job in a JSON file
+    """
+    def __init__(self, dump_location=None):
+        self.errors = []
+        self.dump_location = dump_location
+
+    def parse(self, item):
+        """WIll parse and return the job json
+
+        The received item is a job object with proxy
+        """
+        starting_name, starting_node = "job", item
+        self.job_json = JsonDumper.object_to_json(starting_name, starting_node)
+        
+
+    @staticmethod
+    def object_to_json(name, node):
+        """Will give the attribute information of the provided `node` object as a python dict
+        """
+        node_info = {
+            "name": node._schema.name,
+            "version": node._schema.version.minor,
+            "category": node._schema.category
+        }
+        for attr_name, attr_object in node._schema.allItems():
+            value = getattr(node, attr_name)
+            # if attr_name == "time":
+                # node_info[attr_name] = JsonDumper.handleDatetime(attr_name, getattr(node, attr_name))
+            if isType(value, (list, tuple, GangaList)):
+                node_info[attr_name] = list(value)        
+            elif isinstance(value, GangaObject):
+                node_info[attr_name] = JsonDumper.object_to_json(attr_name, getattr(node, attr_name))
+
+            # ForReview : We could use pickle to save the datetime
```

Comment:
We have (tried to) avoid pickle in the past as there is no promise that an object pickled with version X of python can be read by version Y.